qiskit/__init__.py

```diff
@@ -83,6 +95,12 @@ def __bool__(self):
                 from qiskit.providers import aer
 
                 self.aer = aer.Aer
+                warnings.warn(
+                    "The qiskit.Aer entry point is deprecated instead you should use this "
+                    "directly from the root of the qiskit-aer package.",
+                    DeprecationWarning,
+                    stacklevel=2,
+                )
```

Comment:
We're only releasing the `qiskit-aer` package shortly after this release of Terra.  Perhaps we should hold off on the deprecation warning til Aer 0.12 - the "must support two versions" rule probably applies to Aer here, not Terra.  If we were pointing people to `qiskit.providers.aer` we could do it in this release, but this message probably needs `qiskit-aer` to have existed for longer.

Reply:
I think `qiskit.Aer` is probably so heavily used we should definitely err on the side of caution here.  Let's push the public message one more release.